src/parsing.js

```diff
@@ -102,10 +102,83 @@
     return matchedUnits;
   }
 
+  function ignore(value) {
+    return function(input) {
+      var result = value(input);
+      if (result)
+        result[0] = undefined;
+      return result;
+    }
+  }
+
+  // Regex should be anchored with /^
+  function consumeToken(regex, string) {
+    var result = regex.exec(string);
+    if (result) {
+      result = regex.ignoreCase ? result[0].toLowerCase() : result[0];
+      return [result, string.substr(result.length)];
```

Comment:
I think that'll make the code pretty verbose, but not strongly opposed? What if I just add a comment saying that the parseX functions all return [consumed, remainder] at the top?